rubin_sim/scheduler/basis_functions/basis_functions.py

```diff
@@ -2004,3 +2005,34 @@ def _calc_value(self, conditions, indx=None):
             if self.survey_features["N_obs_survey"].feature > 0
             else 1
         )
+
+
+class RewardNObsSequence(Base_basis_function):
+    """Reward taking a sequence of observations.
+
+    Parameters
+    ----------
+    n_obs_survey : int
+        Number of observations to reward.
+    note_survey : str
+        The value of the observation note, to take into account.
+    nside : int, optional
+        Healpix map resolution (ignored).
+
+    Notes
+    -----
+    This basis function is usefull when a survey is composed of more than one
+    observation (e.g. in different filters) and one wants to make sure they are
+    all taken together.
+    """
+
+    def __init__(self, n_obs_survey, note_survey, nside=None):
+        super().__init__(nside=nside)
+
+        self.n_obs_survey = n_obs_survey
+
+        self.survey_features = {}
+        self.survey_features["N_obs_survey"] = features.N_obs_survey(note=note_survey)
+
+    def _calc_value(self, conditions, indx=None):
+        return self.survey_features["N_obs_survey"].feature % self.n_obs_survey
```

Comment:
How does this reward the observations in a survey being taken together? Maybe I am just not familiar with the current reward functions, etc. n_obs_survey = 'number of observations to reward' .. I would have thought this meant that the observations were sequential or maybe that once you had this many observations in this survey, that that number of observations was what you wanted to reward? but I am not sure?)
Maybe what I'm looking for here is a way to understand how to combine this with other reward functions, such as slew time, etc, if you're coming into the scheduler and seeing this function. 
(note: I think this kind of information is probably lacking all over the scheduler, but is something that seems like it would be helpful to start including).